### PR TITLE
dev-qt/qtnetworkauth: new package (tech preview)

### PR DESCRIPTION
**Pull request by kenya888**
QtNetworkAuth comes from Qt 5.9 (TP), developed on 5.8 branch also.

### CODE REVIEW

**Comment by kenya888** [comment]
travis-ci log:
>   media-gfx/phototonic/phototonic-9999.ebuild: Ebuild uses http:// but should use https:// on line: 14
mmm... what should I do?

**Comment by Chiitoo** [comment]
I believe the 'media-gfx/phototonic/phototonic' issue is unrelated to your changes.

The actual issue seems to be the 'dependency.bad' hits, which in turn are due to the 5.8.0 packages being masked in 'profiles/package.mask' (I think).

**Comment by Pesa** [comment]
> The actual issue seems to be the 'dependency.bad' hits, which in turn are due to the 5.8.0 packages being masked in 'profiles/package.mask' (I think).

Correct. But please forget about 5.8.x, we will be dropping those ebuilds soon and jump directly to 5.9.0

File: dev-qt/qtnetworkauth/qtnetworkauth-5.9999.ebuild [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+inherit qt5-build
+
+DESCRIPTION="Network authorization component library for the Qt5 framework"
+LICENSE="GPL-3"
```
Comment (Pesa) [inline dev-qt/qtnetworkauth/qtnetworkauth-5.9999.ebuild]: are you sure about this?

File: dev-qt/qtnetworkauth/qtnetworkauth-5.9999.ebuild [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+if [[ ${QT5_BUILD_TYPE} == release ]]; then
+	KEYWORDS="~amd64 ~x86"
+fi
+
+IUSE=""
+
+DEPEND="
+	~dev-qt/qtnetwork-${PV}
```
Comment (Pesa) [inline dev-qt/qtnetworkauth/qtnetworkauth-5.9999.ebuild]: missing `qtcore` dep

File: dev-qt/qtnetworkauth/qtnetworkauth-5.9999.ebuild [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+"
+RDEPEND="${DEPEND}"
+
+src_prepare() {
```
Comment (Pesa) [inline dev-qt/qtnetworkauth/qtnetworkauth-5.9999.ebuild]: don't define `src_prepare()` if you're only calling the eclass function.

File: dev-qt/qtnetworkauth/qtnetworkauth-5.9999.ebuild (hunk 1 above)
Comment (kenya888) [inline dev-qt/qtnetworkauth/qtnetworkauth-5.9999.ebuild]: Sorry, could you let me know more about your question?
It's no problem if you concern the license. qtnetworkauth is published under GPL-3 [1]

[1] http://code.qt.io/cgit/qt/qtnetworkauth.git/tree/?h=5.9

File: dev-qt/qtnetworkauth/qtnetworkauth-5.9999.ebuild (hunk 2 above)
Comment (kenya888) [inline dev-qt/qtnetworkauth/qtnetworkauth-5.9999.ebuild]: Oh, OK.

File: dev-qt/qtnetworkauth/qtnetworkauth-5.9999.ebuild (hunk 3 above)
Comment (kenya888) [inline dev-qt/qtnetworkauth/qtnetworkauth-5.9999.ebuild]: I see.

**Comment by kenya888** [comment]
@Chiitoo Thank you for your support.
@Pesa Thank you for your review. I'd like to update the change.

**Comment by kenya888** [comment]
@Pesa Is it better for us to wait pushing this change without 5.8.x files after 5.9 is officially released?

File: dev-qt/qtnetworkauth/qtnetworkauth-5.9999.ebuild (hunk 1 above)
Comment (Pesa) [inline dev-qt/qtnetworkauth/qtnetworkauth-5.9999.ebuild]: yeah I was asking about the license. You're right though, qtnetworkauth seems to be GPL3-only, unlike most other qt modules.

**Comment by Pesa** [comment]
> @Pesa Is it better for us to wait pushing this change without 5.8.x files after 5.9 is officially released?

No, I'm going to drop 5.8.x as soon as next week I think. People can use the 5.9.0 betas if they want.

File: dev-qt/qtnetworkauth/qtnetworkauth-5.9999.ebuild (hunk 1 above)
Comment (kenya888) [inline dev-qt/qtnetworkauth/qtnetworkauth-5.9999.ebuild]: Just for a reference, qtcharts has been also published under GPL-3 only. [2]
I refer the ebuild file of qtcharts for this change.
Could you let me know if I should care more about the license?

[2] http://code.qt.io/cgit/qt/qtcharts.git/tree/?h=5.9

**Comment by kenya888** [comment]
> No, I'm going to drop 5.8.x as soon as next week I think. People can use the 5.9.0 betas if they want.

OK, I continue working. Thanks.

File: dev-qt/qtnetworkauth/qtnetworkauth-5.9999.ebuild (hunk 1 above)
Comment (kenya888) [inline dev-qt/qtnetworkauth/qtnetworkauth-5.9999.ebuild]: I refer the ebuild file of qtcharts, that is published under GPL-3 [2]
Could you let me know if I should care more about the license?

[2] http://code.qt.io/cgit/qt/qtcharts.git/tree/

**Comment by Pesa** [comment]
Squashed, removed 5.8.x ebuilds, and merged. Thanks a lot!

**Comment by kenya888** [comment]
Thanks!!

**Pull request closed without merging**